compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -138,12 +140,16 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 assert!(src.layout.is_sized());
                 assert!(dest.layout.is_sized());
                 if src.layout.size != dest.layout.size {
+                    let src_bytes = src.layout.size.bytes();
+                    let dest_bytes = dest.layout.size.bytes();
+                    let src_ty = format!("{}", src.layout.ty);
+                    let dest_ty = format!("{}", dest.layout.ty);
                     throw_ub_format!(
-                        "transmuting from {}-byte type to {}-byte type: `{}` -> `{}`",
-                        src.layout.size.bytes(),
-                        dest.layout.size.bytes(),
-                        src.layout.ty,
-                        dest.layout.ty,
+                        fluent::const_eval_invalid_transmute,
```

Comment:
If this has a dedicated message now, why should it still use the `Custom` variant?

Or conversely, if `Custom` variants are now 'typed' like this, then why do we need the other variants at all?

Also, for Miri we can still use arbitrary strings here, right?

Reply:
I'm fine with leaving FIXMEs, I was just trying to understand why the code looks the way it does. If anything I feel it'd make sense to move more messages to `Custom` than the other way around, but I might be missing some parts of the trade-off here.